packages/ember/src/components/form/radio-group.gts

```diff
@@ -0,0 +1,109 @@
+import { hash } from '@ember/helper';
+import { on } from '@ember/modifier';
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+
+import BoundValue from './bound-value.ts';
+
+import type { Optional } from '../../types';
+import type { ComponentLike } from '@glint/template';
+
+export interface RadioGroupFieldSignature {
+  Element: HTMLDivElement;
+  Args: {
+    name: string;
+    basic?: boolean;
+    disabled?: boolean;
+    onChange?: (value: string, ...args: unknown[]) => void;
+  };
+  Blocks: {
+    default: [{ Radio: ComponentLike<RadioField> }];
+  };
+}
+
+export default class RadioGroupField extends BoundValue<
+  RadioGroupFieldSignature,
+  string
+> {
+  @action
+  change(updatedValue: Optional<string>) {
+    if (updatedValue) {
+      this.value = updatedValue;
+      this.args.onChange?.(updatedValue);
+    }
+  }
+
+  get classList() {
+    const classes = ['form-control'];
+
+    if (this.args.basic) {
+      classes[0] += '-plaintext';
+    }
+
+    return classes.join(' ');
+  }
+
+  <template>
+    <div class={{this.classList}} ...attributes>
+      {{yield
+        (hash
+          Radio=(component
+            RadioField name=@name disabled=@disabled onChange=this.change
```

Comment:
I'm not entirely sure how this is working currently.
Without `@currentValue` bound it shouldn't react to being checked.

Reply:
The value is updated but it wouldn't set on first load. We have no form for that do that currently. I have added it.